### PR TITLE
Compare raw attributes only when ensuring uniqueness of paginated related models

### PR DESCRIPTION
**Pull request by jaulz**
- [ ] Added or updated tests
- [ ] Documented user facing changes
- [x] Updated CHANGELOG.md

**Changes**

Right now, `toJson` is used which "is recursive, so all attributes and relations will be converted to JSON" (see https://laravel.com/docs/11.x/eloquent-serialization#serializing-to-json). This means that all accessors are executed as well which can affect the performance of an application. The new approach simply compares the raw attributes of the models which should be enough in almost all cases as far as I can see.

**Breaking changes**

There might be a tiny chance that something doesn't work as before because accessors are not compared anymore.

### CODE REVIEW

File: src/Execution/ModelsLoader/PaginatedModelsLoader.php [hunk 1]
```diff
@@ -84,7 +85,9 @@ protected function loadRelatedModels(EloquentCollection $parents): EloquentColle
         return $relatedModels->unique(
             // Compare all attributes because there might not be a unique primary key
             // or there could be differing pivot attributes.
-            static fn (Model $relatedModel): string => $relatedModel->toJson(),
+            static function (Model $relatedModel): string {
+                return json_encode($relatedModel->getOriginal());
+            },
```
Comment (spawnia) [inline src/Execution/ModelsLoader/PaginatedModelsLoader.php:90]: I don't see the need for a full closure, a short fn is simpler here

File: src/Execution/ModelsLoader/PaginatedModelsLoader.php [hunk 2]
```diff
@@ -84,7 +85,9 @@ protected function loadRelatedModels(EloquentCollection $parents): EloquentColle
         return $relatedModels->unique(
             // Compare all attributes because there might not be a unique primary key
             // or there could be differing pivot attributes.
```
Comment (spawnia) [inline src/Execution/ModelsLoader/PaginatedModelsLoader.php:87]: Are pivot attributes also included in `getOriginal()`?

Comment (jaulz) [reply src/Execution/ModelsLoader/PaginatedModelsLoader.php:87]: Yes, they are:
image

File: src/Execution/ModelsLoader/PaginatedModelsLoader.php (hunk 1 above)
Comment (jaulz) [reply src/Execution/ModelsLoader/PaginatedModelsLoader.php]: Fixed

**Comment by skaesdorf** [comment]
@spawnia Hi, this breaks when Models have native Enum casted attributes. `$model->getOriginal()` casts the attributes. `json_encode($relatedModel->getOriginal())` will then fail because enums are not stringable.

**Comment by jaulz** [comment]
Hm, the best would actually be to pick `$original` from the model but it's protected.

**Comment by skaesdorf** [comment]
That would be `$model->getRawOriginal()`

**Comment by jaulz** [comment]
@skaesdorf great, can you raise a PR for that?

**Pull request merged**